tests/Fixer/ControlStructure/NoUnneededControlParenthesesFixerTest.php

```diff
@@ -430,125 +344,1221 @@ function foo() { $a = (yield($x)); }
+            '<?php $aj = [1,("".foo(1+2)),3];',
+        ];
+
+        yield 'block type index square brace' => [
+            '<?php $n = $foo[1];',
+            '<?php $n = $foo[(1)];',
+        ];
+
+        yield 'multiple wrapped call' => [
+            '<?php
+                $u = $z($b);
+                $a = bar/*1*/ ($a) /*2*/;',
+            '<?php
+                $u = $z(($b));
+                $a = bar/*1*/ (((($a)))) /*2*/;',
+        ];
+
+        yield 'if' => [
+            '<?php if ($z) {echo 1;}',
+            '<?php if ((($z))) {echo 1;}',
+        ];
+
+        yield 'destructuring square brace + comments' => [
+            '<?php
+                [/*A*/$a/*B*/] = z();
+                [/*A*/$a/*B*/, $c] = z();
+            ',
+            '<?php
+                [/*A*/($a)/*B*/] = z();
+                [/*A*/($a)/*B*/,($c)] = z();
+            ',
+        ];
+
+        yield 'multiple fix cases' => [
+            '<?php
+                $aF=array(1);$a=array(1);$a=array(1);$a=array(1);
+                $bF=array(1);$b=array(1);$b=array(1);$b=array(1);
+                $cF=array(1);$c=array(1);$c=array(1);$c=array(1);
+            ',
+            '<?php
+                $aF=array((1));$a=array((1));$a=array((1));$a=array((1));
+                $bF=array((1));$b=array((1));$b=array((1));$b=array((1));
+                $cF=array((1));$c=array((1));$c=array((1));$c=array((1));
+            ',
+        ];
+
+        yield 'multiple fix cases with space inserts' => [
+            '<?php $a=$b. 1 .$c;$a=$b. 1 .$c;$a=$b. 1 .$c;$a=$b. 1 .$c;$a=$b. 1 .$c;$a=$b./*1*/1/*2*/.$c;',
+            '<?php $a=$b.(1).$c;$a=$b.(1).$c;$a=$b.(1).$c;$a=$b.(1).$c;$a=$b.(1).$c;$a=$b./*1*/(1)/*2*/.$c;',
+        ];
+
+        yield 'empty exit/die' => [
+            '<?php exit; die;',
+            '<?php exit(); die();',
+        ];
+
+        yield 'more space around concat handling' => [
+            '<?php
+                $s2 = "a". 0.1 . 2 .    "b"    . "c";
+                $f = \'x\' .  \'y\'  . \'z\';
+            ',
+            '<?php
+                $s2 = "a".(0.1). (2) .    ("b")    . "c";
+                $f = \'x\' . ( \'y\' ) . \'z\';
+            ',
+        ];
+
+        $assignOperators = [
+            'and equal' => '&=',
+            'coalesce_equal' => '??=',
+            'concat_equal' => '.=',
+            'div equal' => '*=',
+            'minus equal' => '!=',
+            'mod equal' => '+=',
+            'mul equal' => '*=',
+            'or equal' => '+=',
+            'plus equal' => '+=',
+            'pow equal' => '**=',
+            'sl equal' => '<<=',
+            'sr equal' => '>>=',
+            'xor equal' => '^=',
+        ];
+
+        foreach ($assignOperators as $assignOperator) {
+            yield [
+                '<?php $a '.$assignOperator.' $x;',
+                '<?php $a '.$assignOperator.' ($x);',
+            ];
+        }
+
+        yield 'do not fix' => [
```

Comment:
I've put these in the `do not fix` because I think these cases should not ever be fixed.
While we know that these `(` are currently `)` overcomplete, there is the explicit no BC promise from PHP to consider here I think:
```
Operator precedence and associativity only determine how expressions are grouped, they do not specify an order of evaluation. PHP does not (in the general case) specify in which order an expression is evaluated and code that assumes a specific order of evaluation should be avoided, because the behavior can change between versions of PHP or depending on the surrounding code.
```

https://www.php.net/manual/en/language.operators.precedence.php

I think this can be discussed in a follow up PR if someone really wants this to change :+1:

Reply:
> Operator precedence and associativity only determine how expressions are grouped, they do not specify an order of evaluation.

What does that mean? Aren't they grouped in order to be evaluated according to these groupings?